config_src/drivers/timing_tests/time_MOM_EOS.F90

```diff
@@ -0,0 +1,205 @@
+
+implicit none
+
+! This macro is used to write out timings of a single test rather than conduct
+! a suite of tests. It is not meant for general consumption.
+#undef PDF_ONLY
+
+integer, parameter :: n_fns = 4
+character(len=40) :: fn_labels(n_fns)
+
+! Testing parameters:
+!  nic is number of elements to compute density for (array size), per call
+!  halo is data on either end of the array that should not be used
+!  nits is how many times to repeat the call between turning the timer on/off
+!       to overcome limited resolution of the timer
+!  nsamp repeats the timing to collect statistics on the measurement
+#ifdef PDF_ONLY
+integer, parameter :: nic=26, halo=4, nits=10000, nsamp=400
+#else
+integer, parameter :: nic=23, halo=4, nits=1000, nsamp=400
+#endif
+
+real :: times(nsamp) ! For observing the PDF
+
+! Arrays to hold timings:
+!  first axis corresponds to the form of EOS
+!  second axis corresponds to the function being timed
+real, dimension(:,:), allocatable :: timings, tmean, tstd, tmin, tmax
+integer :: n_eos, i, j
+
+n_eos = size(list_of_eos)
+allocate( timings(n_eos,n_fns), tmean(n_eos,n_fns) )
+allocate( tstd(n_eos,n_fns), tmin(n_eos,n_fns), tmax(n_eos,n_fns) )
+
+fn_labels(1) = 'calculate_density_scalar()'
+fn_labels(2) = 'calculate_density_array()'
+fn_labels(3) = 'calculate_spec_vol_scalar()'
+fn_labels(4) = 'calculate_spec_vol_array()'
+
+tmean(:,:) = 0.
+tstd(:,:) = 0.
+tmin(:,:) = 1.e9
+tmax(:,:) = 0.
+do i = 1, nsamp
+#ifdef PDF_ONLY
+  call run_one(list_of_EOS, nic, halo, nits, times(i))
+#else
+  call run_suite(list_of_EOS, nic, halo, nits, timings)
+  tmean(:,:) = tmean(:,:) + timings(:,:)
+  tstd(:,:) = tstd(:,:) + timings(:,:)**2 ! tstd contains sum or squares here
+  tmin(:,:) = min( tmin(:,:), timings(:,:) )
+  tmax(:,:) = max( tmax(:,:), timings(:,:) )
+#endif
+enddo
+tmean(:,:) = tmean(:,:) / real(nsamp)
+tstd(:,:) = tstd(:,:) / real(nsamp) ! convert to mean of squares
+tstd(:,:) = tstd(:,:) - tmean(:,:)**2 ! convert to variance
+tstd(:,:) = sqrt( tstd(:,:) * ( real(nsamp) / real(nsamp-1) ) ) ! Standard deviation
+
+#ifdef PDF_ONLY
+open(newunit=i, file='times.txt', status='replace', action='write')
+write(i,'(1pE9.3)') times(:)
+close(i)
+#else
+
+! Display results in YAML
+write(*,'(a)') "{"
+do i = 1, n_eos
+  do j = 1, n_fns
+    write(*,"(2x,5a)") '"MOM_EOS_', trim(get_EOS_name(id=list_of_EOS(i))), ' ', trim(fn_labels(j)), '": {'
+    write(*,"(4x,a,1pe11.4,',')") '"min": ',tmin(i,j)
+    write(*,"(4x,a,1pe11.4,',')") '"mean":',tmean(i,j)
+    write(*,"(4x,a,1pe11.4,',')") '"std": ',tstd(i,j)
+    write(*,"(4x,a,i7,',')") '"n_samples": ',nsamp
+    if (i*j.ne.n_eos*n_fns) then
+      write(*,"(4x,a,1pe11.4,'},')") '"max": ',tmax(i,j)
+    else
+      write(*,"(4x,a,1pe11.4,'}')") '"max": ',tmax(i,j)
+    endif
+  enddo
+enddo
+write(*,'(a)') "}"
+#endif
+
+contains
+
+subroutine run_suite(EOS_list, nic, halo, nits, timings)
+integer, intent(in)  :: EOS_list(n_eos) !< IDs of EOS forms to loop over
```

Comment:
The contents of the subroutines `run_suite()` and `run_one()` should probably be indented by two spaces to follow the patterns used elsewhere in the MOM6 code.